### PR TITLE
Chore/offchain secrets readme

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

**Review by pinebit — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -161,3 +164,11 @@ Client contracts which initiate a request and receive a fulfillment can be modif
 ## Simulating Requests
 
 An end-to-end request initiation and fulfillment can be simulated using the `npx hardhat functions-simulate` command. This command will report the total estimated cost of a request in LINK using the latest on-chain gas prices. Costs are based on the amount of gas used to validate the response and call the client contract's `fulfillRequest` function, plus a flat fee. Please note that actual request costs can vary based on gas prices when a request is initiated on-chain.
+
+## Off-chain Secrets
+
+Instead of using encrypted secrets stored directly on the blockchain, encrypted secrets can also be hosted on URLs and be fetched by the DON nodes when a request is initiated.  To generate the encrypted secrets JSON file, run the command `npx hardhat functions-build-offchain-secrets --network network_name_here`.  This will output the file `offchain-secrets.json` which can be hosted at any URL such as Github or S3.
```
Comment (bolekk) [inline README.md:170]: Files are not "hosted on URLs", URLs point to hosting locations. Reword to just "hosted off-chain" and then at the end "hosted anywhere (e.g. Github, S3, etc)."

Also add that currently HTTP is the only supported protocol.

File: README.md [hunk 2]
```diff
@@ -161,3 +164,11 @@ Client contracts which initiate a request and receive a fulfillment can be modif
...
+Then, set `secretsLocation` to `Location.Remote` in `Functions-request-config.js` and enter the URL(s) where the secrets are hosted into `secretsURLs`.  Multiple URLs can be entered as a fallback in case any of the URLs are offline.  Each URL should host the exact same JSON file.  The tooling will automatically pack the secrets URL(s) into a space-separated string and encrypt the string using the DON public key so no 3rd party can view the URLs.  Finally, this encrypted string of URLs is used in the `secrets` parameter when making an on-chain request.
```
Comment (bolekk) [inline README.md:171]: Starting with "Then" feels like I missed some first step already. Maybe start with "In order to use off-chain secrets, ..."

File: README.md [hunk 3]
```diff
@@ -161,3 +164,11 @@ Client contracts which initiate a request and receive a fulfillment can be modif
...
+
+Additionally, off-chain secrets enable a separate set of secrets to be assigned to each node in the DON.  Each node will not be able to decrypt the set of secrets belonging to another node.  Optionally, a set of default secrets encrypted with the DON public key can be used as a fallback by any DON member who does not have a set of secrets assigned to them.  This handles the case where a new member is added to the DON, but the assigned secrets have not yet been updated.
+To use per-node assigned secrets, enter a list of objects into `perNodeSecrets` in `Functions-request-config.js` before running the `functions-build-offchain-secrets` command.  The number of objects in the array must correspond to the number of nodes in the DON.  Default secrets can be entered into the `secrets` parameter of `Functions-request-config.js`.  Each object must have the same set of entries, but the values for each entry can be different (ie: `[ { apiKey: '123' }, { apiKey: '456' } ]`).
```
Comment (bolekk) [inline README.md:174]: You could add that the URLs need to be accessible when request is being processed. However, we recommend that the URLs expire when they are no longer in use.

**Review by bolekk — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (KuphJr) [reply README.md:170]: Improved wording

**Review by bolekk — APPROVED** [review]

**Pull request merged**